pkg/resources/kafka/kafka.go

```diff
@@ -1203,20 +1218,50 @@ func getServiceFromExternalListener(client client.Client, cluster *v1beta1.Kafka
 // - prioritize upscale in order to allow upscaling the cluster even when there is a stuck RU
 // - prioritize missing broker pods to be able for escaping from offline partitions, not all replicas in sync which
 //		could stall RU flow
-func reorderBrokers(brokerPods corev1.PodList, desiredBrokers []v1beta1.Broker, brokersState map[string]v1beta1.BrokerState, controllerBrokerID int32) []v1beta1.Broker {
+func reorderBrokers(brokerPods corev1.PodList, desiredBrokers []v1beta1.Broker, brokersState map[string]v1beta1.BrokerState, pvcList corev1.PersistentVolumeClaimList, controllerBrokerID int32, log logr.Logger) []v1beta1.Broker {
 	runningBrokers := make(map[string]struct{})
 	for _, b := range brokerPods.Items {
 		brokerID := b.GetLabels()["brokerId"]
 		runningBrokers[brokerID] = struct{}{}
 	}
 
+	presentPersistentVolumeClaims := make(map[string]struct{})
+	for _, pvc := range pvcList.Items {
+		brokerID := pvc.GetLabels()["brokerId"]
+		presentPersistentVolumeClaims[brokerID] = struct{}{}
```

Comment:
Instead of initializing `presentPersistentVolumeClaims` here do it outside of `reorderBrokers` and pass in as parameter from outside.

Reply:
You're right the `runnigBrokers` should be passed in from outside as well. (e.g. `runnigBrokers sets.String` and `brokersWithPVCs sets.String`